spec/index.html

```diff
@@ -1516,58 +1516,78 @@ <h3>The <code>rdf:JSON</code> Datatype</h3>
-        [[RFC8259]].</dd>
+        <a data-cite="RFC8259#section-2">Section&nbsp;2 JSON Grammar</a> of [[RFC8259]],
+        which also conform to the requirements of [[[RFC7493]]] [[RFC7493]].
+        <div class="note">
+          [[[RFC8259]]] [[RFC8259]] allows strings to include
+          <a data-cite="I18N-GLOSSARY#dfn-surrogate" class="lint-ignore">surrogate code points</a>
+          not allowed in <a data-lt="string">RDF strings</a>,
+          which are also excluded in [[RFC7493]],
+          thus the lexical representation of JSON literals excludes those including
+          <a data-cite="I18N-GLOSSARY#dfn-surrogate" class="lint-ignore">surrogate code points</a>.
+        </div></dd>
 
       <dt id="JSON-value-space">The <a>value space</a></dt>
-      <dd>
-        is the set of all <a data-lt="string">RDF strings</a> that conform to the
-        <a data-cite="RFC8259#section-2">JSON Grammar</a> as described in
-        <a data-cite="RFC8259#section-2">Section&nbsp;2 JSON Grammar</a> of
-        [[RFC8259]], and furthermore comply with the following constraints:
+      <dd>is the set of
+        <a>strings</a>,
+        numbers (<a data-cite="XMLSCHEMA11-2#double"><strong>xsd:double</strong></a>),
+        <a data-cite="INFRA#ordered-map">maps (unordered)</a>,
+        <a data-cite="INFRA#list">lists</a>, and
+        literal values (<a data-cite="INFRA#boolean">`true`, `false`</a>, and <a data-cite="INFRA#nulls">`null`</a>)
+        from [[[INFRA]]] [[INFRA]] and [[[XMLSCHEMA11-2]]] [[XMLSCHEMA11-2]].
+
+        <p>Two values (|a| and |b|) are considered equal if:
+          <ul>
+            <li>they are the same <a>string</a>,
+              number (<a data-cite="XMLSCHEMA11-2#double"><strong>xsd:double</strong></a>), or
+              literal value;</li>
+            <li>they are both <a data-cite="INFRA#list">lists</a> containing <a data-cite="INFRA#list-item">items</a>
+              which are pairwise equal – meaning that each <a data-cite="INFRA#list-item">item</a>
+              in |a| is equal the <a data-cite="INFRA#list-item">item</a>
+              at the corresponding index in |b|,
+              and both |a| and |b| have the same <a data-cite="INFRA#list-size">size</a>; or</li>
+            <li>if they are both <a data-cite="INFRA#ordered-map">maps</a> with equal <a data-cite="INFRA#entry">entries</a>
+              – meaning that both |a| and |b| have the same <a data-cite="INFRA#map-size">size</a>,
+              and for each entry <var>e<sub>a</sub></var> in |a|
+              there is an entry <var>e<sub>b</sub></var> in |b|
+              such that the <a data-cite="INFRA#map-key">key</a> in <var>e<sub>a</sub></var>
+              equals the <a data-cite="INFRA#map-key">key</a> in <var>e<sub>b</sub></var>,
+              and the <a data-cite="INFRA#map-value">value</a> in <var>e<sub>a</sub></var>
+              equals the <a data-cite="INFRA#map-value">value</a> in <var>e<sub>b</sub></var>.</li>
```

Comment:
```suggestion
              equals the <a data-cite="INFRA#map-value">value</a> in <var>e<sub>b</sub></var>.
              <div class="note">Two JSON Objects containing map entries which are serialized in a different order will be equal under this definition.
              For example, `{ "a": 1, "b": 2} and {"b":2, "a":1}` are considered equal.
              </div>
            </li>
```

Reply:
These two literals are definitely not equal, nor would they be if they were rdf:JSON literals.   What are equal (and identical, etc.) are the result of the lexical-to-value map for rdf:JSON and this must be true in the value space for rdf:JSON and this is not evident in the current wording.

Again, it is much better to have the value space be correct from the beginning instead of having an ugly ambiguous patch over an incorrect value space.  Either a source should be chosen that has a correct definition or a definition should be given in the document.   (Well, it would be even better if JSON had a suitable value space but that ship sailed very, very long ago.)

Reply:
I made a tweak to the language to clarify that it is the values which are equal, not the original literal representations.

Without creating a new definition for unordered map, which I don't think is advisable. Based on feedback, we could reconsider this, but I think the president for using INFRA for describing JSON values in other W3C specs has been established.